method/scanpy/method_scanpy.py

```diff
@@ -90,36 +90,44 @@
 # ... or AnnData if you want
 def get_anndata(args):
     import anndata as ad
+    import numpy as np
+    import pandas as pd
     import scipy as sp
     from PIL import Image
 
-    X = sp.io.mmread(args.matrix)
-    if sp.sparse.issparse(X):
-        X = X.tocsr()
-
     observations = pd.read_table(args.observations, index_col=0)
     features = pd.read_table(args.features, index_col=0)
 
-    # Filter
-    if "selected" in observations.columns:
-        X = X[observations["selected"].to_numpy().nonzero()[0], :]
-        observations = observations.loc[lambda df: df["selected"]]
-    if "selected" in features.columns:
-        X = X[:, features["selected"].to_numpy().nonzero()[0]]
-        features = features.loc[lambda df: df["selected"]]
-
     coordinates = (
         pd.read_table(args.coordinates, index_col=0)
         .loc[observations.index, :]
         .to_numpy()
     )
 
-    adata = ad.AnnData(
-        X=X, obs=observations, var=features, obsm={"spatial_pixel": coordinates}
-    )
+    adata = ad.AnnData(obs=observations, var=features, obsm={"spatial": coordinates})
+
+    if args.matrix is not None:
+        X = sp.io.mmread(args.matrix)
+        if sp.sparse.issparse(X):
+            X = X.tocsr()
+        adata.X = X
+
+    if args.neighbors is not None:
+        adata.obsp["spatial_connectivities"] = sp.io.mmread(args.neighbors).T.tocsr()
+
+    # Filter by selected samples/features
+    if "selected" in adata.obs.columns:
+        adata = adata[observations["selected"].astype(bool), :]
+    if "selected" in adata.var.columns:
+        adata = adata[:, features["selected"].astype(bool)]
+
+    if args.dim_red is not None:
+        adata.obsm["reduced_dimensions"] = (
+            pd.read_table(args.dim_red, index_col=0).loc[adata.obs_names].to_numpy()
+        )
 
-    if args.image is not None:
-        adata.uns["image"] = np.array(Image.open(args.image))
+    if args.img is not None:
+        adata.uns["image"] = np.array(Image.open(args.img))
```

Comment:
Why this change?

Reply:
> Why this change?

This is the way it is written in the original template "method.py" in the SpaceHack2. For some reason, I had "image" written out, but in your template it is "img"

Reply:
Four days and you are the first to notice that the template is wrong! It actually has to be `args.image` if you look at the CLI. Can you change it back. And sorry for the wrong template

Reply:
> Four days and you are the first to notice that the template is wrong! It actually has to be `args.image` if you look at the CLI. Can you change it back. And sorry for the wrong template

Haha nice! No worries at all, I am glad we sorted it out:)